lib/Frontend/Frontend.cpp

```diff
@@ -402,10 +402,12 @@ static void addAdditionalInitialImportsTo(
 }
 
 static bool shouldImportSwiftOnoneModuleIfNoneOrImplicitOptimization(
-    FrontendOptions::ActionType RequestedAction) {
+    FrontendOptions::ActionType RequestedAction, bool TrackSystemDeps) {
   return RequestedAction == FrontendOptions::ActionType::EmitObject ||
          RequestedAction == FrontendOptions::ActionType::Immediate ||
-         RequestedAction == FrontendOptions::ActionType::EmitSIL;
+         RequestedAction == FrontendOptions::ActionType::EmitSIL ||
+         (TrackSystemDeps
+          && RequestedAction == FrontendOptions::ActionType::ResolveImports);
```

Comment:
Is this even the right condition?  Should we *always* force this load when `TrackSystemDeps` is true?

Reply:
Another option would be to manually insert this dependency into the tracker if `TrackSystemDeps` is true but this condition in its original form returns false.

Reply:
The problem for me is that the comment on the TrackSystemDeps option doesn't explain enough about the "why". Not your fault, is preexisting. But I cannot tell if this condition is correct from the comments.